kratos/utilities/math_utils.h

```diff
@@ -1215,43 +1215,46 @@ class MathUtils
      * \f$ [ s11, s22, s12 ] \f$ for 2D case.
      * @param rStressVector the given stress vector
      * @return the corresponding stress tensor in matrix form
+     * @tparam TVector The vector type considered
+     * @tparam TMatrixType The matrix returning type
      */
-    static inline MatrixType StressVectorToTensor(const Vector& rStressVector)
+   template<class TVector, class TMatrixType = MatrixType>
```

Comment:
Also, to maintain compatibility, shouldn't be:

```C++
template<class TVector = Vector, class TMatrix = MatrixType>
```

And also, to keep the style, if we call a template `TVector` the other type should be `TMatrix`, or `TVectorType` and `TMatrixType`

Reply:
No, because the first type is explictely detected

Reply:
Implicitily

Reply:
But we are adding the possibility to use this function with any type having [] operator. 

For example I can define 

```C++ 
class foo {
    int& operator[] (int x) {
        return 0;
    }
}
``` 

As a vector, and it will be perfectly fine to call it with this new class. Not saying its wrong, but I am not sure we intend to open the function this way.